uk_results/migrations/0030_populate_postresult_post_election.py

```diff
@@ -0,0 +1,88 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+from collections import defaultdict
+
+from django.db import migrations, models
+
+
+def set_post_election_from_post(apps, schema_editor):
+    """
+    This is far from idea. Try to guess the PostExtraElection
+    that this PostResult relates to. This will have to be done by looking
+    and the related memberships and assuming they're correct (sometimes they
+    won't be, and that will have to be fixed manually later).
```

Comment:
Ah, thanks to your tip I see [how the`CandidateResult`s are being created](https://github.com/DemocracyClub/yournextrepresentative/blob/b5e26a2d67d8c5def4de58280106f318a8dccd29/uk_results/forms.py#L284-L289). So when filling in results at the moment, people are shown a form which one row for any candidacy associated with that post (i.e. from any  election). So we have to hope that if people see a mix of candidates from both elections, they've stopped and not filled anything in (and not "corrected" data for a candidacy from the last election), but that might have happen.

I think I get that now :) I do think that, once you're done with fixing these immediate problems, someone writing a script that takes an old database dump and looks for discrepancies with the fixed data would be a good idea, anyway.